src/main/java/fr/openent/competences/service/impl/DefaultCompetenceNoteService.java

```diff
@@ -343,6 +346,31 @@ public void getConversionNoteCompetence(String idEtablissement, String idClasse,
         Sql.getInstance().prepared(query.toString(), values, SqlResult.validResultHandler(handler));
     }
 
+    public Future<JsonArray> getConversionNoteCompetence(String idEtablissement, String idClasse) {
+
+        Promise<JsonArray> conversionTable = Promise.promise();
+        JsonArray values = new fr.wseduc.webutils.collections.JsonArray();
+        StringBuilder query = new StringBuilder()
+                .append("SELECT valmin, valmax, coalesce(perso.libelle, niv.libelle) as libelle, ordre, niv.couleur, bareme_brevet ")
+                .append("FROM notes.niveau_competences AS niv ")
+                .append("INNER JOIN  ").append(schema).append("echelle_conversion_niv_note AS echelle ON niv.id = echelle.id_niveau ")
+                .append("INNER JOIN  ").append(schema).append("rel_groupe_cycle CC ON cc.id_cycle = niv.id_cycle ")
+                .append("AND cc.id_groupe = ? ")
+                .append("AND echelle.id_structure = ? ")
+                .append("LEFT JOIN (SELECT * FROM ").append(schema).append("perso_niveau_competences ")
+                .append("WHERE id_etablissement = ?) AS perso ON (perso.id_niveau = niv.id) ")
+                .append("ORDER BY  ordre DESC");
+        values.add(idClasse);
+        values.add(idEtablissement);
+        values.add(idEtablissement);
+        Sql.getInstance().prepared(query.toString(), values,
+                SqlResult.validResultHandler(FutureHelper.handlerJsonArray(conversionTable,
+                        "[DefaultCompetenceNoteService] : getConversionNoteCompetence")));
+
+        return conversionTable.future();
+    }
```

Comment:
Même principe par rapport à mon commentaire au-dessus : utilise la méthode déjà existante pour polymorpher ta méthode que tu veux faire retourner une future 

Exemple usage : 

```suggestion
 public Future<JsonArray> getConversionNoteCompetence(String idEtablissement, String idClasse) {
        Promise<JsonArray> conversionTable = Promise.promise();
        getConversionNoteCompetence(idEtablissement, idClasse, event -> {
            if (event.isLeft()) {
                log.error(String.format("[Competences@%s::getConversionNoteCompetence] Error during sql request: %s.",
                        this.getClass().getSimpleName(), event.left().getValue()));
                conversionTable.fail(event.left().getValue());
            } else {
                conversionTable.complete(event.right().getValue());
            }
        });
        return conversionTable.future();
    }
```